api/src/test/java/org/apache/commons/rdf/api/AbstractGraphTest.java

```diff
@@ -228,6 +231,128 @@ public void getTriplesQuery() throws Exception {
         assertEquals(3, graph.getTriples(null, member, null).count());
     }
 
+    @Test
+    public void addBlankNodesFromMultipleGraphs() {
+
+        try {
+            Graph g1 = createGraph1();
+            Graph g2 = createGraph2();
+            Graph g3 = factory.createGraph();
+
+            copyTriples(g1, g3);
+            copyTriples(g2, g3);
+
+            IRI name = factory.createIRI("http://xmlns.com/foaf/0.1/name");
+
+            final Map<String, BlankNodeOrIRI> whoIsWho = new HashMap<>();
+            System.out.println(g3.getTriples(null, name, null).count());
+            // NOTE: sequential as our HashMap is not thread-safe
```

Comment:
This test plays safe and simple.. it's like 6 triples - yet without .sequential() I got buggy results now and then. :)  Similarly I made the copyTriple sequential as I can't know what the tested Graph supports.

Expected synchronization details (or their explicit lack of) for all of Graph should be discussed as a new issue.